src/Umbraco.Web/Editors/CodeFileController.cs

```diff
@@ -306,7 +306,7 @@ public HttpResponseMessage Delete(string type, string virtualPath)
             switch (type)
             {
                 case Core.Constants.Trees.PartialViews:
-                    if (IsDirectory(virtualPath, SystemDirectories.PartialViews))
+                    if (IsDirectory(virtualPath, SystemDirectories.PartialViews, Current.FileSystems.PartialViewsFileSystem))
```

Comment:
Can't we assume that if the `IFileService` can't find the partial view/file, it should be a folder, so we can remove the `IsDirectory()` method and use this instead:
```suggestion
                    if (Services.FileService.GetPartialView(virtualPath) == null)
```

This would remove the requirement get the `IFileSystem` from the `Current` singleton as well 👌🏻

Reply:
I guess we could, however, this would cause us to load in the contents of the files, just to then delete them, which to me seems pretty inefficient and slow, compared to the current solution.

Reply:
Valid point... I noticed `DeletePartialView()` returns a bool, so another option would be to do the following:
```c#
if (Services.FileService.DeletePartialView(virtualPath, Security.CurrentUser.Id))
{
    return Request.CreateResponse(HttpStatusCode.OK);
}
else
{
    Services.FileService.DeletePartialViewFolder(virtualPath);
    return Request.CreateResponse(HttpStatusCode.OK);
}
```

If it's a file, it gets deleted right away (without fetch it - including any content - first) and otherwise it tries to delete the folder (which silently ignores non-existent paths). It doesn't return the `NotFound` error though, but that's an edge case and not finding it while you're trying to delete it anyway shouldn't be an issue, right?

Reply:
Or more concise:
```c#
if (Services.FileService.DeletePartialView(virtualPath, Security.CurrentUser.Id) == false)
{
    Services.FileService.DeletePartialViewFolder(virtualPath);
}

return Request.CreateResponse(HttpStatusCode.OK);
```

Reply:
That would would work for partial view perhaps, but `DeleteStylesheet` and `DeleteScript`, doesn't return a bool, so it wouldn't work

Reply:
Right, so I wrongfully assumed to having a consistent API 😞

My main concern is that we're now decoupling the `IFileService` from the physical file system, but coupling it to specific `IFileSystem`s again... In that regard, the minor overhead of fetching the stylesheets/scripts before deleting is simply the cost to pay for not having an API capable of exposing the right information.